Document OS error "Bad message" unhealthy case (#30851)
home-assistant/home-assistant.github.io: source/more-info/unhealthy/oserror_bad_message.markdown

---
title: "Operating System error: Bad message"
description: "More information on what to do when OS errors mark the installation as unhealthy."
---

## The issue

> The Supervisor received Operating System errors during file operation (Python OSError exceptions). The error number reported was 74, "Bad message", which is typically raised on file system corruption.

Reviewer: ```suggestion
The Supervisor received Operating System errors during file operation (Python OSError exceptions). The error number reported was 74, "Bad message", which is typically raised on file system corruption. File system corruption can happen due to a power outage or when the system otherwise did not get shut down properly. 
```

## The solution

> File system corruption can happen due to power outage or when the system otherwise did not get shutdown properly. Typically Home Assistant recovers from file system corruption, so if you see this error the first time consider just ignoring it. Just in case the problem gets worse, make sure to create a full backup (and download it or store it on a different system).

Reviewer: ```suggestion
Typically, Home Assistant recovers from file system corruption. If you see this error for the first time, ignore it. In case the problem gets worse, make sure to create a full backup, download it, or store it on a different system.
```

> If this issue reappears even after a system restart, your system configuration or hardware might be at fault. Consider a clean reinstallation of your Operating System. If you are using a externally attached USB drive, consider replacing the USB adapter or the SSD. If you are confident that the hardware is not at fault and you are using a Raspberry Pi, it might be related to incompatibility with certain UAS drives (this [Raspberry Pi forum post][rpi-forum-uas] has some more background information). Home Assistant OS maintains a list of devices where UAS gets disabled to work around the problem (check config.txt on the first partition of your installation). If the USB PID/VID of your device is missing consider [opening an issue][haos-issue] in the Home Assistant OS project.

Reviewer: ```suggestion
If this issue reappears even after a system restart, your system configuration or hardware might be at fault. Consider a clean reinstallation of your Operating System. If you are using an external USB drive, consider replacing the USB adapter or the SSD. If you are confident that the hardware is not at fault and you are using a Raspberry Pi, it might be related to incompatibility with certain UAS drives. For more background information, read this [Raspberry Pi forum post][rpi-forum-uas]. The Home Assistant OS maintains a list of devices where UAS are disabled in order to work around the problem (check config.txt on the first partition of your installation). If the USB PID/VID of your device is missing, consider [opening an issue][haos-issue] in the Home Assistant OS project.
```
